types/scene/wlr_scene.c

```diff
@@ -1422,6 +1562,59 @@ static void scene_entry_render(struct render_list_entry *entry, const struct ren
 			wlr_output_transform_invert(scene_buffer->transform);
 		transform = wlr_output_transform_compose(transform, data->transform);
 
+		// Blur
+		if (scene_buffer_should_blur(scene_buffer->backdrop_blur, &scene->blur_data)) {
+			pixman_region32_t opaque_region;
+			pixman_region32_init(&opaque_region);
+
+			bool has_alpha = false;
+			struct wlr_scene_surface *scene_surface;
+			if (scene_buffer->opacity < 1.0) {
+				has_alpha = true;
+				pixman_region32_union_rect(&opaque_region, &opaque_region, 0, 0, 0, 0);
+			} else if ((scene_surface = wlr_scene_surface_try_from_buffer(scene_buffer))) {
+				has_alpha = !scene_surface->surface->opaque;
+				pixman_region32_copy(&opaque_region, &scene_surface->surface->opaque_region);
+			} else {
+				has_alpha = pixman_region32_not_empty(&opaque);
+				scene_node_opaque_region(node, x, y, &opaque_region);
+				scale_output_damage(&opaque_region, data->scale);
+			}
```

Comment:
can't we use `scene_node_opaque_region()`?

Reply:
Good catch!